src/metadata.jl

```diff
@@ -0,0 +1,124 @@
+import ProtocolBuffers
+
+# Conversion to Julia primitive types
+# ========================================================================================
+const proto_to_julia_type_map = Dict(
+    Protocol_PB.PrimitiveType.INT_128 => Int128,
+    Protocol_PB.PrimitiveType.INT_64 => Int64,
+    Protocol_PB.PrimitiveType.INT_32 => Int32,
+    Protocol_PB.PrimitiveType.INT_16 => Int16,
+    Protocol_PB.PrimitiveType.INT_8 => Int8,
+    Protocol_PB.PrimitiveType.UINT_128 => UInt128,
+    Protocol_PB.PrimitiveType.UINT_64 => UInt64,
+    Protocol_PB.PrimitiveType.UINT_32 => UInt32,
+    Protocol_PB.PrimitiveType.UINT_16 => UInt16,
+    Protocol_PB.PrimitiveType.UINT_8 => UInt8,
+    Protocol_PB.PrimitiveType.FLOAT_64 => Float64,
+    Protocol_PB.PrimitiveType.FLOAT_32 => Float32,
+    Protocol_PB.PrimitiveType.FLOAT_16 => Float16,
+    Protocol_PB.PrimitiveType.STRING => String,
+    Protocol_PB.PrimitiveType.SYMBOL => Symbol,
+    Protocol_PB.PrimitiveType.CHAR => Char,
+    Protocol_PB.PrimitiveType.BOOL => Bool,
+)
+
+function primitive_type_from_proto(primitive_type::Protocol_PB.PrimitiveType.T)
+    # Each of these maps to a single Julia type.
+    proto_to_julia_type_map[primitive_type]
+end
+
+function extract_values_from_proto(rel_tuple::Protocol_PB.RelTuple)
+    return [value_from_proto(val) for val in rel_tuple.arguments]
+end
+
+# Julia values from protos.
+function value_from_proto(v::Protocol_PB.PrimitiveValue)
+    # Use the `tag` to infer the Julia type.
+    T = primitive_type_from_proto(v.tag)
+
+    # Construct the Julia type.
+    return _from_proto(T, v.value.value)
+end
+
+_from_proto(T::Type, v) = T(v)
+# TODO (dba) Consider creating a `Symbol` here instead of a `String` as in Julia a
+# specialized, or constant, string is a `Symbol`
+# Byte array must be copied because Symbol takes ownership!
+_from_proto(::Type{String}, v) = String(copy(v))
+_from_proto(::Type{Int16}, v) = v % Int16
+_from_proto(::Type{Int8}, v) = v % Int8
+_from_proto(::Type{UInt16}, v) = v % Int16
+_from_proto(::Type{UInt8}, v) = v % Int8
+_from_proto(T::Type{UInt128}, v) = _from_uint128_proto(v)
+_from_proto(T::Type{Int128}, v) = _from_int128_proto(v)
+
+function _from_uint128_proto(i::Protocol_PB.RelUInt128)
+    return (UInt128(i.highbits) << 64) | i.lowbits
+end
+
+function _from_int128_proto(i::Protocol_PB.RelInt128)
+    return (Int128(i.highbits) << 64) | i.lowbits
+end
+
+# Display
+# ========================================================================================
+function show_relation_id(io::IO, rel_id::Protocol_PB.RelationId)
+    for rel_type in rel_id.arguments
+        print(io, "/")
+        show_rel_type(io, rel_type)
+    end
+    print(io, "\n")
+    return nothing
+end
```

Comment:
This is very nifty! 😁 👍 

Personally, i think that the `/`s are a bit out-dated, and we should just leave them behind entirely. They don't correspond to anything in Rel, they're a bit distracting, and they don't stand out compared to the whitespace in an example like this:
`/ConstantType(Symbol, :output)/Int64`, where the whitespace is a more obvious separator than the `/`!

I would suggest that we just format this as a normal-looking julia array, like this:
```julia
[ConstantType(Symbol, :output), Int64]
```
Another nice thing about this format is that we could make it an actually valid copy/pastable representation, I think?

Reply:
Although it's maybe then weird that the `ConstantType(Symbol, :output)` struct doesn't actually exist. ... I was going to suggest what if we _made_ it exist, and we could maybe print it like:
```julia
[ConstantType{Symbol}(:output), Int64]
```
But i think it's maybe not useful to have both the proto structs _and_ separate julia structs that we convert the proto structs into? 🤔 i'm not sure; i'd need to think more about it.

Anyway, maybe what you have here is already good for now, and we can think more about it as we go on.

Maybe we could just drop the whitespace in there? Or maybe _add_ whitespace between the `/`s?:
- `/ ConstantType(Symbol, :output) / Int64`
- `/ConstantType(Symbol,:output)/Int64`

Reply:
See https://relationalai.slack.com/archives/C02C9AV2LL9/p1661963216091409